kong/pdk/vault.lua

```diff
@@ -1264,14 +1264,33 @@ local function new(self)
     local ttl = SECRETS_CACHE:ttl(new_cache_key)
     if ttl and SECRETS_CACHE:get(new_cache_key) ~= NEGATIVELY_CACHED_VALUE then
       local resurrect_ttl = max(config.resurrect_ttl or DAO_MAX_TTL, SECRETS_CACHE_MIN_TTL)
+      -- the secret is still within ttl, no need to refresh
       if ttl > resurrect_ttl then
         return true
+
+      elseif ttl > SECRETS_CACHE_MIN_TTL then
```

Comment:
Not sure I'm understanding this correctly, but why should `ttl > SECRETS_CACHE_MIN_TTL` here?

Reply:
Right now the vault timer is refreshing all the secrets in shared dict with a period that equals to `ROTATION_INTERVAL`, and `SECRETS_CACHE_MIN_TTL` is `2 * ROTATION_INTERVAL`. This ensures the execution of the secret refreshing will happen at least once within the last remaining time of `SECRETS_CACHE_MIN_TTL`, so that it can be refreshed either with a null placholder(`NEGATIVELY_CACHED_VALUE`) or a new updated value.